### PR TITLE
[#542]: Add numbering of copies and tooltips for hovering snippet cards

### PR DESCRIPTION
**Pull request by sergiozeppo**
Hi there! 
Fix for #542 
Fix for #541 

Decided to do both parts of the issue since they both significantly improve UX. 

1. Fixed the function of naming created duplicate snippets. Now the function will not duplicate the infinite insertion of “-copy”, and will take into account the number of copies. 
изображение

2. Also I added tooltips for hovering on snippet cards. There is it's looks like
изображение

### CODE REVIEW

File: frontend/src/utils/genDuplicateSnippetName.js [hunk 1]
```diff
@@ -2,19 +2,25 @@ import { SNIPPET_NAME_MAX_LENGTH } from './validationSchemas';
...
-  } else {
-    const [name, extension] = [
-      fileName.slice(0, extensionIndex),
-      fileName.slice(extensionIndex + 1),
-    ];
-
-    finalName = `${name}-copy.${extension}`;
-  }
+  if (copyTextIndex !== -1) {
+    if (dashIndex > copyTextIndex) {
```
Comment (HelenOne) [inline frontend/src/utils/genDuplicateSnippetName.js:14]: if inside other if isnt a good practice. Maybe you can replace it with sequential checks?

File: frontend/src/pages/profile/SnippetCardWrapper.jsx [hunk 2]
```diff
@@ -1,5 +1,13 @@
 function SnippetCardWrapper({ children }) {
-  return <div className="snippet-card-wrapper">{children}</div>;
+  const hoverTooltip =
+    children.props.children.length > 1
+      ? children.props.children[0].props.data?.name
```
Comment (HelenOne) [inline frontend/src/pages/profile/SnippetCardWrapper.jsx:4]: Not readable. It is better to put a part of the logic into a variable first

**Review by HelenOne — CHANGES_REQUESTED** [review]
Added some comments

File: frontend/src/pages/profile/SnippetCardWrapper.jsx (hunk 2 above)
Comment (sergiozeppo) [reply frontend/src/pages/profile/SnippetCardWrapper.jsx]: > Not readable. It is better to put a part of the logic into a variable first

Fixed.

File: frontend/src/utils/genDuplicateSnippetName.js (hunk 1 above)
Comment (sergiozeppo) [reply frontend/src/utils/genDuplicateSnippetName.js]: > if inside other if isnt a good practice. Maybe you can replace it with sequential checks?

Hi, Helen! 
Fixed nested conditions. Check, please.

**Comment by sergiozeppo** [comment]
> Added some comments

Hi, @HelenOne ! 
Thanks a lot for your valuable remarks! 

Sorry for being so long inactive in fixing this tasks, was busy. 
Now I think it become better. 
Check it. when you'll find the free time.

**Pull request merged**